.github/workflows/build.yml

```diff
@@ -174,6 +179,9 @@ jobs:
     - name: ouroboros-network [build]
       run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-network
 
+    - name: ouroboros-consensus [build]
+      run: cabal --builddir="$CABAL_BUILDDIR" build ouroboros-consensus
+
```

Comment:
*If* we decide to build consensus in GitHub Actions, we should build all of it, not just one package. A manual topological sort of the dependencies:

* `ouroboros-consensus`
* `ouroboros-consensus-mock`
* `ouroboros-consensus-test`
* `ouroboros-consensus-mock-test`
* `ouroboros-consensus-byron`
* `ouroboros-consensus-byronspec`
* `ouroboros-consensus-byron-test`
* `ouroboros-consensus-shelley`
* `ouroboros-consensus-shelley-test`
* `ouroboros-consensus-cardano`
* `ouroboros-consensus-cardano-test`

All these steps are sequential right? Can't we parallelise them? For example, why not `cabal build all` and then `cabal test all`? Is that just to see which component failed?

BTW, I'm not convinced that we should include consensus. We already have Hydra, why add another kind of CI? For network, it makes sense, as they want to test Windows-only code, which is not possible with Hydra. But for consensus, there is hardly any Windows-only code.
I guess one reason would be to test with GHC 8.6.5? One could say that the CI in `cardano-node` covers this already.

Reply:
`ghc-8.6.5` is a good reason.  I think the one if `cardano-node` doesn't cover this because it is in a separate repository and won't prevent code that doesn't compile on `ghc-8.6.5` from merging in this one.

It possible to only build `consensus` in Linux only as well.

Reply:
I think the reason we don't do `build all` is because the repository includes `Win32-network` which doesn't build on POSIX.  That project really ought to be in its own repository.

Reply:
@newhoggy I glanced at the `Win32-network.cabal` and it looks like it disables most things when not built on windows. Moreover, `cabal v2-build all` works in my `nix-shell`.

Could you point me towards one of the `Win32-network`-on-POSIX build errors?

Reply:
It actually builds now:

https://github.com/input-output-hk/ouroboros-network/pull/2868/checks?check_run_id=1697932929

Should we just switch to doing that?

Reply:
Nice! :+1

Yeah, I think `v2-build all` is appropriate and will save some maintenance.

Reply:
Build all in done.